presto-iceberg/src/main/java/com/facebook/presto/iceberg/nessie/IcebergNessieSessionProperties.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg.nessie;
+
+import com.facebook.presto.cache.CacheConfig;
+import com.facebook.presto.hive.OrcFileWriterConfig;
+import com.facebook.presto.hive.ParquetFileWriterConfig;
+import com.facebook.presto.iceberg.IcebergConfig;
+import com.facebook.presto.iceberg.IcebergSessionProperties;
+import com.facebook.presto.spi.session.PropertyMetadata;
+import com.google.common.collect.ImmutableList;
+
+import javax.inject.Inject;
+
+import static com.facebook.presto.iceberg.nessie.IcebergNessieCatalogFactory.NESSIE_REFERENCE_HASH;
+import static com.facebook.presto.iceberg.nessie.IcebergNessieCatalogFactory.NESSIE_REFERENCE_NAME;
+import static com.facebook.presto.spi.session.PropertyMetadata.stringProperty;
+
+public final class IcebergNessieSessionProperties
+        extends IcebergSessionProperties
+{
+    @Inject
+    public IcebergNessieSessionProperties(
+            IcebergConfig icebergConfig,
+            ParquetFileWriterConfig parquetFileWriterConfig,
+            OrcFileWriterConfig orcFileWriterConfig,
+            CacheConfig cacheConfig,
+            IcebergNessieConfig catalogConfig)
+    {
+        super(icebergConfig, parquetFileWriterConfig, orcFileWriterConfig, cacheConfig);
+        sessionProperties = ImmutableList.<PropertyMetadata<?>>builder()
```

Comment:
Yeah I went back and forth on this part of the re-factor several times trying to determine the best way. In the very next line of this code block, it's adding all the superclass properties into its own `sessionProperties` instance, and I haven't seen any adverse effects. That being said, happy to hear out suggestions on a better way for this

Reply:
Personally I think it's more concise and clearer to use `IcebergSessionProperties` directly. If we only want to bind `IcebergNessieConfig` in `IcebergNessieCatalogModule` when catalogType is NESSIE, then we may consider define an optional injection point by `newOptionalBinder(binder, IcebergNessieConfig.class)` in `IcebergCommonModule` or `IcebergCatalogModule`, so that we can accept an `Optional<IcebergNessieConfig>` directly in `IcebergSessionProperties`. In that way, we can remove this `IcebergNessieSessionProperties`. 

Bring up this just for discussion. What's you opinion, did I overlook any important reason?

Reply:
I like that idea! I was trying to really strictly enforce the separation of catalogs, but it was just a bit too heavy-duty of an approach when it came to the SessionProperties class, as evidenced by the fact that there wasn't really an elegant way to handle it. I like the simplicity of this approach better and incorporated it into the latest